src/yarp.c

```diff
@@ -2224,7 +2237,16 @@ parse_parameters(yp_parser_t *parser) {
         local.end -= 1;
         yp_token_list_append(&parser->current_scope->as.scope.locals, &local);
 
-        yp_node_t *param = yp_node_keyword_parameter_node_create(parser, &name);
+	yp_node_t *default_value = NULL;
+	if (parser->current.type != YP_TOKEN_NEWLINE && // TODO: do we need terminator here?
+	    parser->current.type != YP_TOKEN_SEMICOLON &&
+	    parser->current.type != YP_TOKEN_COMMA &&
+	    parser->current.type != YP_TOKEN_PARENTHESIS_RIGHT &&
+	    parser->current.type != YP_TOKEN_EOF) {
+	  default_value = parse_expression(parser, BINDING_POWER_NONE, "Expected a default value after keyword argument which does not terminate.");
+	}
+
+        yp_node_t *param = yp_node_keyword_parameter_node_create(parser, &name, default_value);
```

Comment:
Are you ok reusing the `keyword_parameter` node here? I am not sure how isolated parameter and argument are intended to be from each other.